extension/src/openvic-extension/singletons/CursorSingleton.cpp

```diff
@@ -0,0 +1,586 @@
+		Variant::ARRAY,
+		"cursor_names", PROPERTY_HINT_ARRAY_TYPE,
+		"String"),
+	"", "get_cursor_names");
+}
+
+CursorSingleton* CursorSingleton::get_singleton() {
+	return _singleton;
+}
+
+CursorSingleton::CursorSingleton() {
+	ERR_FAIL_COND(_singleton != nullptr);
+	_singleton = this;
+}
+
+CursorSingleton::~CursorSingleton() {
+	ERR_FAIL_COND(_singleton != this);
+	_singleton = nullptr;
+}
+
+
+Array CursorSingleton::get_frames(String const& name, int res_index){
+	if(res_index < cursors[name].images.size()){
+		return cursors[name].images[res_index];
+	}
+	return Array();
+}
+
+TypedArray<Vector2i> CursorSingleton::get_hotspots(String const& name, int res_index){
+	if(res_index < cursors[name].images.size()){
+		return cursors[name].hotspots[res_index];
+	}
+	return TypedArray<Vector2i>();
+}
+
+int CursorSingleton::get_animationLength(String const& name){
+	return cursors[name].animationLength;
+}
+
+TypedArray<Vector2i> CursorSingleton::get_resolutions(String const& name){
+	return cursors[name].resolutions;
+}
+
+TypedArray<float> CursorSingleton::get_displayRates(String const& name){
+	if(cursors[name].displayRates.has_value()){
+		return cursors[name].displayRates.value();
+	}
+	return TypedArray<float>();
+}
+
+TypedArray<int> CursorSingleton::get_sequence(String const& name){
+	if(cursors[name].displayRates.has_value()){
+		return cursors[name].sequence.value();
+	}
+	return TypedArray<int>();
+}
+
+void CursorSingleton::generate_resolution(String const& name, int base_res_index, Vector2i target_res){
+	cursor_asset_t* cursor = &cursors[name];
+	Array new_frameset;
+	TypedArray<Vector2i> new_hotspots;
+
+	for(int i=0;i<cursor->images[base_res_index].size();i++){
+
+		Ref<ImageTexture> texture = static_cast<Ref<ImageTexture>>(cursor->images[base_res_index][i]);
+		
+		Ref<Image> image;
+		image.instantiate();
+		image->create(target_res.x, target_res.y, false, Image::Format::FORMAT_RGBA8);
+		image->copy_from(texture->get_image());
+		image->resize(target_res.x,target_res.y,Image::INTERPOLATE_BILINEAR);
+
+
+		Ref<ImageTexture> new_texture = Ref<ImageTexture>();
+		new_texture.instantiate();
+		new_texture->set_image(image);
+
+		new_frameset.push_back(new_texture);
+
+		Vector2i base_hotspot = cursor->hotspots[base_res_index][i];
+		Vector2i scale_ratio = target_res / static_cast<Vector2i>(cursor->resolutions[base_res_index]);
+		new_hotspots.push_back(base_hotspot*scale_ratio);
+
+	}
+
+	cursor->images.push_back(new_frameset);
+	cursor->hotspots.push_back(new_hotspots);
+	cursor->resolutions.push_back(target_res);
+}
+
+//, std::string_view const& base_folder
+String CursorSingleton::to_define_file_name(String const& path) const {
+	String name = path.replace("\\","/");
+	return name.get_slice("gfx/cursors/",1).get_slice(".",0);
+}
+
+bool CursorSingleton::load_cursors() {
+	GameSingleton const* game_singleton = GameSingleton::get_singleton();
+	ERR_FAIL_NULL_V_MSG(game_singleton, false, vformat("Error retrieving GameSingleton"));
```

Comment:
`vformat` isn't needed if there's nothing to be substituted in (using the same `%` system as GDScript)